### PR TITLE
Collect data from Host Inventory service and forward to Outlier detection service

### PR DESCRIPTION
**Pull request by jameswnl**
@tumido @AparnaKarve can you help review this?

### CODE REVIEW

**Comment by jameswnl** [comment]
@miq-bot add_reviewer @tumido 
@miq-bot add_reviewer @AparnaKarve

File: workers.py [hunk 1]
```diff
@@ -308,9 +310,56 @@ def worker_topology(topology_info: dict) -> None:
 
         logger.debug('%s: Done, exiting', thread.name)
 
+    def worker_host(info: dict) -> None:
+        """Download and forward the content."""
+        thread = current_thread()
+        logger.debug('%s: Worker started', thread.name)
+
+        identity = json.loads(base64.b64decode(b64_identity))
+        account_id = identity.get('identity').get('account_number')
```
Comment (tumido) [inline workers.py]: Do we need the account Id for anything apart from logging? As a unique identifier, the `b64_identity` would serve the same purpose as well.

Comment (jameswnl) [reply workers.py]: The `account_id` is being used as the [batch_id](https://github.com/ManageIQ/aiops-ai-library/pull/17/files#diff-738ac49b5cddf3f7ce9f1a17ec48f249R56) which is in turn the [upload data id](https://github.com/ManageIQ/aiops-ai-library/pull/17/files#diff-738ac49b5cddf3f7ce9f1a17ec48f249R88) in `outlier-detection` service.

**Comment by jameswnl** [comment]
@tumido this error in travis are strange
```
target_worker/__init__.py:92:80: E501 line too long (88 > 79 characters) [pycodestyle]
target_worker/__init__.py:96:80: E501 line too long (81 > 79 characters) [pycodestyle]
```
This file ends at line 83

**Comment by tumido** [comment]
Hey @jameswnl, can you please separate the `worker_class` related code into a separate PR? This is a separate task [AIOPS-129](https://projects.engineering.redhat.com/browse/AIOPS-129) which is assigned to me and I've planned to work on it after my current task. Feel free to take over (reassign the card), however please make the code as a separate PR.

File: config.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 workers = int(os.environ.get('GUNICORN_PROCESSES', '3'))
 threads = int(os.environ.get('GUNICORN_THREADS', '1'))
+worker_class = os.environ.get('GUNICORN_WORKER', 'sync')
```
Comment (tumido) [inline config.py]: This should be a separate PR

File: workers.py [hunk 3]
```diff
@@ -308,9 +338,49 @@ def worker_topology(topology_info: dict) -> None:
...
```
Comment (tumido) [inline workers.py]: In case the `identity` scheme changes, this line can fail. If `identity` has no key `'identity'` it would return `None`. Calling `.get('account_number')` would result into complete task failure. Let's change this to at least: 
```suggestion
        account_id = identity.get('identity', {}).get('account_number')
```

File: workers.py [hunk 4]
```diff
@@ -168,6 +171,33 @@ def query_sub_collection(
     return all_data
 
 
+def retrieve_hosts(host_inventory_url, b64_identity) -> dict:
+    headers = {"x-rh-identity": b64_identity}
+    url = host_inventory_url + '&page={}'
```
Comment (tumido) [inline workers.py]: What happens if an host is added into the inventory at the time we're listing this? We would get duplicit data on the second page? How do we handle this?

File: tests/test_get_hosts.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import unittest
```
Comment (tumido) [inline tests/test_get_hosts.py]: Let's not use `unittest`. Pytest is prefered, see https://github.com/ManageIQ/aiops-incoming-listener/blob/master/tests/test_app.py

File: workers.py (hunk 3 above)
Comment (jameswnl) [reply workers.py]: My did that previously, but then switched back to allow it to error out.
Yeah, shouldn't assume the scheme.

File: workers.py (hunk 4 above)
Comment (jameswnl) [reply workers.py]: `rad` would deal with duplicates

File: workers.py (hunk 3 above)
Comment (tumido) [reply workers.py]: I mean.. we shouldn't care what's inside the `b64_identity`. That's platform thing. It's our own initiate we are decoding it. But our service shouldn't fail if there's a sudden change in the scheme. We're not the ones whose setting the rules, so we wouldn't be even told if there's change happening.... So we should rely on it.

Comment (jameswnl) [reply workers.py]: > Yeah, shouldn't assume the scheme.

by that (and my latest push) I meant an agreement  😉

File: tests/test_get_hosts.py (hunk 5 above)
Comment (jameswnl) [reply tests/test_get_hosts.py]: @tumido is this a blocker for this PR? `pytest` looks great, I'll follow up with another PR for that.
Would like to get his in as it's blocking an end to end trip of the outlier detection pipeline
